python/cuml/preprocessing/TargetEncoder.py

```diff
@@ -0,0 +1,213 @@
+        Default number of folds
+    smooth : int (default=0)
+        Number of samples to smooth the encoding
+    seed : int (default=42)
+        Random seed
+    split : {'random', 'continuous', 'interleaved'}, default='interleaved'
+    Examples
+    --------
+    Converting a categorical implementation to a numerical one
+
+    .. code-block:: python
+
+        from cudf import DataFrame, Series
+
+        train = DataFrame({'category': ['a', 'b', 'b', 'a'],
+                           'label': [1, 0, 1, 1]})
+        test = DataFrame({'category': ['a', 'c', 'b', 'a']})
+
+        # There is only one correct way to do this
+        encoder = TargetEncoder()
+        train_encoded = encoder.fit_transform(train.category, train.label)
+        test_encoded = encoder.transform(test.category)
+        print(train_encoded)
+        print(test_encoded)
+
+    Output:
+
+    .. code-block:: python
+
+        [1. 1. 0. 1.]
+        [1.   0.75 0.5  1.  ]
+
+    """
+    def __init__(self, n_folds=4, smooth=0, seed=42, split='interleaved'):
+        self.n_folds = n_folds
+        self.seed = seed
+        self.smooth = smooth
+        self.split = split
+        self.y_col = '__TARGET__'
+        self.x_col = '__FEA__'
+        self.out_col = '__TARGET_ENCODE__'
+        self.fold_col = '__FOLD__'
+        self.id_col = '__INDEX__'
+        self.train = None
+
+    def fit(self, x, y):
+        res, train = self._fit_transform(x, y)
+        self.train_encode = res
+        self.train = train
+        self._fitted = True
+        return self
+
+    def fit_transform(self, x, y):
+        self.fit(x, y)
+        return self.train_encode
+
+    def transform(self, x):
+        self._check_is_fitted()
+        test = self._to_frame(x)
+        if self._is_train_df(test):
+            return self.train_encode
+        x_cols = [i for i in test.columns.tolist() if i != self.id_col]
+        test = test.merge(self.agg_all, on=x_cols, how='left')
+        return self._get_return_value(test)
+
+    def _fit_transform(self, x, y):
+        cp.random.seed(self.seed)
+        train = self._to_frame(x)
+        x_cols = [i for i in train.columns.tolist() if i != self.id_col]
+
+        if isinstance(y, gd.Series):
+            train[self.y_col] = y.values
+        elif isinstance(y, cp.ndarray):
+            if len(y.shape) == 1:
+                train[self.y_col] = y
+            elif y.shape[1] == 1:
+                train[self.y_col] = y[:, 0]
+            else:
+                raise ValueError(f"Input of shape {y.shape} "
+                                 "is not a 1-D array.")
+        else:
+            raise TypeError(
+                "Input of type {type(y)} is not cudf.Series, "
+                "or cupy.ndarray")
+
+        if self.split == 'random':
+            train[self.fold_col] = cp.random.randint(0,
+                                                     self.n_folds, len(train))
+        elif self.split == 'continuous':
```

Comment:
not sure I understand the continuous split... the other two are clear, maybe just describe in the docstring

Reply:
yep, added in docstring.